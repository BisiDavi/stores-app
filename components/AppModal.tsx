import React, {PropsWithChildren} from 'react';
import {View} from 'react-native';
import Modal from 'react-native-modal';

export default function AppModal({
  children,
  visible,
  style,
}: PropsWithChildren<AppModalProps>) {
  return (
    <View>
      <Modal style={style} isVisible={visible}>
        {children}
      </Modal>
    </View>
  );
}

interface AppModalProps {
  toggleOverlay: () => void;
  visible: boolean;
  style: any;
}
