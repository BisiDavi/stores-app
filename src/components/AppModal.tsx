import React, {PropsWithChildren} from 'react';
import {View} from 'react-native';
import Modal from 'react-native-modal';

export default function AppModal({
  children,
  visible,
  style,
  toggleOverlay,
}: PropsWithChildren<AppModalProps>) {
  return (
    <View>
      <Modal style={style} isVisible={visible} onBackdropPress={toggleOverlay}>
        {children}
      </Modal>
    </View>
  );
}

export interface AppModalProps {
  toggleOverlay: () => void;
  visible: boolean;
  style?: any;
}
