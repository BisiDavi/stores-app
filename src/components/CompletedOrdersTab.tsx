import React, {useCallback, useEffect, useState} from 'react';
import {FlatList, TouchableOpacity, View, Text, StyleSheet} from 'react-native';
import {useSelector} from 'react-redux';

import CompletedOrdersList from '@/json/completed-order.json';
import OrdersListItem from '@/components/OrdersListItem';
import {getCompletedOrdersRequest} from '@/network/postRequest';
import {RootState} from '@/store/RootReducer';

export default function CompletedOrdersTab({navigation}: any) {
  const [completedOrder, setCompletedOrder] = useState([]);
  const keyExtractor = useCallback(item => item.id.toString(), []);
  const {storeProfile}: any = useSelector(
    (state: RootState) => state.storeProfile,
  );
  useEffect(() => {
    let once = true;
    getCompletedOrdersRequest({storeId: storeProfile._id}).then(response => {
      if (once) {
        setCompletedOrder(response.data.data);
      }
    });
    return () => {
      once = false;
    };
  }, [storeProfile]);

  return (
    <>
      {completedOrder.length > 0 ? (
        <FlatList
          data={completedOrder}
          renderItem={({item}) => (
            <TouchableOpacity
              onPressIn={() => navigation.navigate('ViewOrderScreen', item)}
            >
              <OrdersListItem item={item} />
            </TouchableOpacity>
          )}
          initialNumToRender={5}
          maxToRenderPerBatch={5}
          keyExtractor={keyExtractor}
        />
      ) : (
        <View style={styles.container}>
          <Text style={styles.text}>
            Hello {storeProfile.name}, you currently don't have any completed
            orders
          </Text>
        </View>
      )}
    </>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    textAlign: 'center',
    fontFamily: 'Roboto-Bold',
    fontSize: 16,
  },
});
