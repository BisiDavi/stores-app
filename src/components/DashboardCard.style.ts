import {Dimensions, StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  dashboardCard: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    margin: 0,
    justifyContent: 'flex-start',
    padding: 0,
  },
  card: {
    height: 80,
    width: Dimensions.get('window').width * 0.38,
    display: 'flex',
    margin: 0,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 0,
  },
  cardText: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 14,
    lineHeight: 16,
  },
  title: {
    fontWeight: '500',
    fontFamily: 'Roboto-Regular',
    fontSize: 14,
    lineHeight: 20,
    marginBottom: 10,
  },
});
