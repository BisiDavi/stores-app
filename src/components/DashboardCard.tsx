import React from 'react';
import {Card} from 'react-native-elements';
import {View, TouchableOpacity, Text} from 'react-native';
import {styles} from './DashboardCard.style';

interface DashboardCardProps {
  content: {
    title: string;
    amount: string;
    link?: string | undefined;
  };
  navigation: any;
}

export default function DashboardCard({
  content,
  navigation,
}: DashboardCardProps) {
  return (
    <View style={styles.dashboardCard}>
      <Text style={styles.title}>{content.title}</Text>
      {content.link ? (
        <TouchableOpacity onPress={() => navigation.navigate(content.link)}>
          <Card containerStyle={styles.card}>
            <Text style={styles.cardText}>{content.amount}</Text>
          </Card>
        </TouchableOpacity>
      ) : (
        <Card containerStyle={styles.card}>
          <Text style={styles.cardText}>{content.amount}</Text>
        </Card>
      )}
    </View>
  );
}
