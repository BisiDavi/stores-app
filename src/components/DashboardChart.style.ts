import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  dashboardChart: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    marginBottom: 15,
  },
  lineChartStyle: {
    marginVertical: 8,
    borderRadius: 16,
    marginLeft: 20,
  },
  chart: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    position: 'relative',
  },
  yAxis: {
    display: 'flex',
    flexDirection: 'column',
    transform: [{rotate: '270deg'}],
    margin: 0,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 0,
    height: 200,
    top: 20,
    position: 'absolute',
    left: 35,
    fontFamily: 'Roboto-Regular',
    fontSize: 16,
    lineHeight: 16,
  },
  xAxis: {
    fontFamily: 'Roboto-Regular',
    fontSize: 16,
    lineHeight: 16,
  },
});
