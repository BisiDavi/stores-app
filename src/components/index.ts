export {default as AppModal} from './Modal/AppModal';
export {default as CompletedOrdersTab} from './Tabs/CompletedOrdersTab';
export {default as DashboardCard} from './DashboardCard';
export {default as InputField} from './InputField';
export {default as LoadingActivityIndicator} from './Loader/LoadingActivityIndicator';
export {default as NewOrdersTab} from './Tabs/NewOrdersTab';
export {default as ProgressIndicator} from './ProgressIndicator/ProgressIndicator';
export {default as RadioField} from './RadioField';
export {default as SelectField} from './FormElements/Select/SelectField';
export {default as Fab} from './Fab';
