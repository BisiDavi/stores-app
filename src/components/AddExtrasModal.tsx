import React from 'react';
import {Dimensions, StyleSheet, Text, View} from 'react-native';
import {Button} from 'react-native-elements';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {useNavigation} from '@react-navigation/core';
import AppModal, {AppModalProps} from './AppModal';
import addExtrasButtons from '@/json/add-extra-modal.json';
import {colors} from '@/utils';

export default function AddExtrasModal({
  visible,
  toggleOverlay,
}: AppModalProps) {
  const navigation: any = useNavigation();
  return (
    <AppModal visible={visible} toggleOverlay={toggleOverlay}>
      <View style={styles.modalView}>
        <TouchableOpacity>
          <Text style={styles.text}>Done</Text>
        </TouchableOpacity>
        <View style={styles.buttonGroup}>
          {addExtrasButtons.map(button => {
            if (button.link) {
              return (
                <Button
                  title={button.text}
                  titleStyle={styles.buttonText}
                  onPress={() => navigation.navigate(button.link)}
                  type="clear"
                />
              );
            } else {
              return (
                <Button
                  titleStyle={styles.buttonText}
                  onPress={toggleOverlay}
                  title={button.text}
                />
              );
            }
          })}
        </View>
      </View>
    </AppModal>
  );
}

const styles = StyleSheet.create({
  text: {
    fontFamily: 'Roboto-Bold',
    color: colors.black,
    textAlign: 'center',
  },
  buttonGroup: {
    marginTop: 30,
  },
  buttonText: {
    fontFamily: 'Roboto-Light',
    textAlign: 'center',
  },
  modalView: {
    backgroundColor: colors.neutralWhite,
    height: 180,
    borderRadius: 10,
    padding: 20,
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: Dimensions.get('window').width * 0.5,
  },
});
