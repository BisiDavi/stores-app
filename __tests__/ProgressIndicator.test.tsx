/**
 * @format
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {ProgressIndicator} from '@/components';

describe('Testing the ProgressIndicator Component', () => {
  it('renders correctly, Snapshot test', () => {
    const tree = renderer
      .create(
        <ProgressIndicator
          title="Step 2: Other Details (fresh food)"
          selected={1}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
