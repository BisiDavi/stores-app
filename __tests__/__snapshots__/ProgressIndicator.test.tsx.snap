// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ProgressIndicator Component renders correctly, Snapshot test 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#F29100",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 18,
        "lineHeight": 28,
        "marginBottom": 10,
        "padding": 0,
      }
    }
  >
    Step 2: Other Details (fresh food)
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginTop": 0,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F29100",
          "borderColor": "#707070",
          "borderWidth": 1,
          "height": 5,
          "margin": 0,
          "width": 150,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#707070",
          "borderWidth": 1,
          "height": 5,
          "margin": 0,
          "width": 150,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#707070",
          "borderWidth": 1,
          "height": 5,
          "margin": 0,
          "width": 150,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#707070",
          "borderWidth": 1,
          "height": 5,
          "margin": 0,
          "width": 150,
        }
      }
    />
  </View>
</View>
`;
